context/src/junit/junit_path.rs

```diff
@@ -0,0 +1,29 @@
+#[cfg(feature = "pyo3")]
+use pyo3::prelude::*;
+#[cfg(feature = "pyo3")]
+use pyo3_stub_gen::derive::gen_stub_pyclass_enum;
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "wasm")]
+use tsify_next::Tsify;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+#[cfg_attr(feature = "pyo3", gen_stub_pyclass_enum, pyclass(eq, eq_int))]
+#[cfg_attr(feature = "wasm", derive(Tsify))]
+pub enum TestRunnerJunitStatus {
+    #[default]
+    Passed,
+    Failed,
+    Flaky,
+}
+
+/// Encapsulates the glob path for a junit and, if applicable, the flakiness already
+/// assigned by the user's test runner. See bazel_bep/parser.rs for more.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct JunitPathWrapper {
+    /// Path or glob pattern to the junit file.
+    pub junit_path: String,
+    /// Refers to an optional status parsed from the test runner's output, before junits have been parsed.
+    pub status: Option<TestRunnerJunitStatus>,
+}
```

Comment:
Related to genericizing this, I think we ought to consider this a `JunitReportStatus` and `JunitReportFileWithStatus`. Test runners can tell us the status, but it should be fairly trivial for us to derive it from the JUnit report by seeing how many failures and reruns there were. We could even record both what the test runner says and what we derive from the JUnit report for the express reason of checking if those agree and processing quarantining as the test runner says (lest our inference is incorrect). Given that, I think we should always be able to populate `status` and if that's not true, I think we should probably consider adding an `Unknown` variant instead of making `status` `Option<T>`. With this, we can then extend our quarantining check for all JUnit reports rather than just what Bazel or other test runners tell us.

Reply:
I added a follow ticket to add more generic support for all JUnit report files:
https://linear.app/trunk/issue/TRUNK-13911/parse-junit-files-and-check-for-reruns-for-quarantining-in-analytics

Reply:
Spoke offline, I will undertake the small refactor/renaming here, and then those tickets will capture the follow-up work for parsing the junits themselves